### PR TITLE
[User Service] Add isAdmin property to User model

### PR DESCRIPTION
**Pull request by brein62**
- Added `isAdmin` property to users.
- Allow admins to adjust users' privileges with a POST request to `/users/:id/privilege` with the following body:
`{ "isAdmin": true/false }`
	- Since only existing admins can adjust users' privileges, if you want to be an admin, tell me xd

### CODE REVIEW

File: user-service/src/routes/userRoute.ts [hunk 1]
```diff
@@ -1,12 +1,19 @@
 import { Router } from 'express';
 
-import { createUser, getAllUsers, getUser, updateUser, deleteUser } from '../controllers/userController';
+import { createUser, getAllUsers, getUser, updateUser, deleteUser, updateUserPrivilege } from '../controllers/userController';
 import { protectRoute } from '../middlewares/protectRoute';
+import { adminProtectRoute } from '../middlewares/adminProtectRoute';
```
Comment (tiongMax) [inline user-service/src/routes/userRoute.ts:5]: i think can combine into 1 import

**Review by tiongMax — COMMENTED** [review]
LGTM!

File: user-service/src/routes/userRoute.ts (hunk 1 above)
Comment (brein62) [reply user-service/src/routes/userRoute.ts]: doneded

**Review by wjayee — APPROVED** [review]
LGTM!

**Pull request merged**